include/swift/Reflection/ReflectionContext.h

```diff
@@ -1492,13 +1602,8 @@ class ReflectionContext
       RecordPtr = RecordObj->Parent;
     }
 
-    // Walk the async backtrace if the task isn't running or cancelled.
-    // TODO: Use isEnqueued from https://github.com/apple/swift/pull/41088/ once
-    // that's available.
-    int IsCancelledFlag = 0x100;
-    int IsRunningFlag = 0x800;
-    if (!(AsyncTaskObj->PrivateStorage.Status.Flags[0] & IsCancelledFlag) &&
-        !(AsyncTaskObj->PrivateStorage.Status.Flags[0] & IsRunningFlag)) {
+    // Walk the async backtrace.
+    if (Info.HasIsRunning && !Info.IsRunning) {
```

Comment:
Should the tool print out a warning or something in the output if `HasIsRunning = false`?

Reply:
Good idea, added that.